commit 6338bb05c15f88e2f4beae296ef389224837758c
Author: Masami Hiramatsu (Google) <mhiramat@kernel.org>
Date:   Mon Dec 12 11:46:44 2022 +0900

    error-injection: remove EI_ETYPE_NONE
    
    Patch series "error-injection: Clarify the requirements of error
    injectable functions".
    
    Patches for clarifying the requirement of error injectable functions and
    to remove the confusing EI_ETYPE_NONE.
    
    
    This patch (of 2):
    
    Since the EI_ETYPE_NONE is confusing type, replace it with appropriate
    errno.  The EI_ETYPE_NONE has been introduced for a dummy (error) value,
    but it can mislead people that they can use ALLOW_ERROR_INJECTION(func,
    NONE).  So remove it from the EI_ETYPE and use appropriate errno instead.
    
    [akpm@linux-foundation.org: include/linux/error-injection.h needs errno.h]
    Link: https://lkml.kernel.org/r/167081319306.387937.10079195394503045678.stgit@devnote3
    Link: https://lkml.kernel.org/r/167081320421.387937.4259807348852421112.stgit@devnote3
    Fixes: 663faf9f7bee ("error-injection: Add injectable error types")
    Signed-off-by: Masami Hiramatsu (Google) <mhiramat@kernel.org>
    Cc: Alexei Starovoitov <alexei.starovoitov@gmail.com>
    Cc: Borislav Petkov (AMD) <bp@alien8.de>
    Cc: Chris Mason <clm@meta.com>
    Cc: Christoph Hellwig <hch@infradead.org>
    Cc: Florent Revest <revest@chromium.org>
    Cc: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
    Cc: Jonathan Corbet <corbet@lwn.net>
    Cc: Josh Poimboeuf <jpoimboe@redhat.com>
    Cc: Kees Cook <keescook@chromium.org>
    Cc: KP Singh <kpsingh@kernel.org>
    Cc: Mark Rutland <mark.rutland@arm.com>
    Cc: Peter Zijlstra <peterz@infradead.org>
    Cc: Steven Rostedt (Google) <rostedt@goodmis.org>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>

diff --git a/lib/error-inject.c b/lib/error-inject.c
index 1afca1b1cdea..32c14770508e 100644
--- a/lib/error-inject.c
+++ b/lib/error-inject.c
@@ -40,7 +40,7 @@ bool within_error_injection_list(unsigned long addr)
 int get_injectable_error_type(unsigned long addr)
 {
 	struct ei_entry *ent;
-	int ei_type = EI_ETYPE_NONE;
+	int ei_type = -EINVAL;
 
 	mutex_lock(&ei_mutex);
 	list_for_each_entry(ent, &error_injection_list, list) {